special-pages/pages/new-tab/app/mock-transport.js

```diff
@@ -256,30 +274,46 @@ export function mockTransport() {
-
                     const count = url.searchParams.get('stats-update-count');
-                    const max = Math.min(parseInt(count || '0'), 10);
-                    if (max === 0) return () => {};
-
-                    let inc = 1;
-                    const int = setInterval(() => {
-                        if (inc === max) return clearInterval(int);
-                        const next = {
-                            ...stats.willUpdate,
-                            trackerCompanies: stats.willUpdate.trackerCompanies.map((x, index) => {
-                                return {
-                                    ...x,
-                                    count: x.count + inc * index,
-                                };
-                            }),
+                    const updateMaxCount = parseInt(count || '0');
+                    if (updateMaxCount === 0) return () => {};
+                    if (statsVariant === 'willUpdate') {
+                        let inc = 1;
+                        const max = Math.min(updateMaxCount, 10);
+                        const int = setInterval(() => {
+                            if (inc === max) return clearInterval(int);
+                            const next = {
+                                ...stats.willUpdate,
+                                trackerCompanies: stats.willUpdate.trackerCompanies.map((x, index) => {
+                                    return {
+                                        ...x,
+                                        count: x.count + inc * index,
+                                    };
+                                }),
+                            };
+                            cb(next);
+                            inc++;
+                        }, 500);
+                        return () => {
+                            clearInterval(int);
                         };
-                        cb(next);
-                        inc++;
-                    }, 500);
-                    return () => {
-                        clearInterval(int);
-                    };
+                    } else if (statsVariant === 'growing') {
+                        const list = stats.many.trackerCompanies;
+                        let index = 0;
+                        const max = Math.min(updateMaxCount, list.length);
+                        const int = setInterval(() => {
+                            if (index === max) return clearInterval(int);
+                            console.log({ index, max });
+                            cb({
+                                trackerCompanies: list.slice(0, index + 1),
+                            });
+                            index++;
+                        }, 200);
+                        return () => {};
+                    } else {
+                        console.log(statsVariant);
+                        return () => {};
+                    }
```

Comment:
This is just about adding more scenarios for testing - none of these make it into the final release.